src/components/tl/cuda/alltoallv/alltoallv_ce.c

```diff
@@ -0,0 +1,405 @@
+        src        = PTR_OFFSET(src, data_displ);
+        data_displ = task->alltoallv_ce.get_offset(
+            task, sync->alltoallv_ce.rdispl_bytes, peer);
+        dst = PTR_OFFSET(task->alltoallv_ce.rbuf, data_displ);
+
+        exec_args.task_type = UCC_EE_EXECUTOR_TASK_TYPE_COPY;
+        exec_args.bufs[0]   = dst;
+        exec_args.bufs[1]   = src;
+        exec_args.count     = data_size;
+        exec_task =
+            &task->alltoallv_ce.exec_task[task->alltoallv_ce.num_posted];
+        status = ucc_ee_executor_task_post(exec, &exec_args, exec_task);
+        if (ucc_unlikely(status != UCC_OK)) {
+            goto exit;
+        }
+        task->alltoallv_ce.num_posted++;
+    }
+
+    for (i = 0; i < team->topo->num_proxies; i++) {
+        if (team->topo->proxies[i].proxy == rank) {
+            psrc      = team->topo->proxies[i].src;
+            pdst      = team->topo->proxies[i].dst;
+            peer_sync = TASK_SYNC(task, psrc);
+            // FIXME: assume the same datatype
+            data_size = task->alltoallv_ce.get_size(
+                task, peer_sync->alltoallv_ce.sbytes, pdst);
+            data_displ = task->alltoallv_ce.get_offset(
+                task, peer_sync->alltoallv_ce.sdispl_bytes, pdst);
+            src        = PTR_OFFSET(task->alltoallv_ce.peer_map_addr_src[psrc],
+                                    peer_sync->mem_info_src.offset);
+            src        = PTR_OFFSET(src, data_displ);
+            peer_sync  = TASK_SYNC(task, pdst);
+            dst        = PTR_OFFSET(task->alltoallv_ce.peer_map_addr_dst[pdst],
+                                    peer_sync->mem_info_dst.offset);
+            data_displ = task->alltoallv_ce.get_offset(
+                task, peer_sync->alltoallv_ce.rdispl_bytes, psrc);
+            dst                 = PTR_OFFSET(dst, data_displ);
+            exec_args.task_type = UCC_EE_EXECUTOR_TASK_TYPE_COPY;
+            exec_args.bufs[0]   = dst;
+            exec_args.bufs[1]   = src;
+            exec_args.count     = data_size;
+            exec_task =
+                &task->alltoallv_ce.exec_task[task->alltoallv_ce.num_posted];
+            status = ucc_ee_executor_task_post(exec, &exec_args, exec_task);
+            if (ucc_unlikely(status != UCC_OK)) {
+                goto exit;
+            }
+            task->alltoallv_ce.num_posted++;
+        }
+    }
+    status = ucc_ec_event_post(team->stream, task->alltoallv_ce.copy_done,
+                               UCC_EE_CUDA_STREAM);
+exit:
+    return status;
+}
+
+void ucc_tl_cuda_alltoallv_ce_progress(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+    ucc_status_t        status;
+    int                 i;
+
+    switch (task->alltoallv_ce.stage) {
+    case ALLTOALL_CE_STAGE_SYNC:
+        if (ucc_tl_cuda_get_sync(task) != UCC_OK) {
+            task->super.super.status = UCC_INPROGRESS;
```

Comment:
need to replace all task->super.super.status = something with task->super.status = something, because of this change https://github.com/openucx/ucc/pull/441